Core/Core/Planner/CalendarFilter/View/CalendarFilterScreen.swift

```diff
@@ -0,0 +1,153 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2024-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import SwiftUI
+
+public struct CalendarFilterScreen: View, ScreenViewTrackable {
+    public var screenViewTrackingParameters: ScreenViewTrackingParameters { viewModel.pageViewEvent }
+
+    @ObservedObject private var viewModel: CalendarFilterViewModel
+    @Environment(\.viewController) private var viewController
+
+    public init(viewModel: CalendarFilterViewModel) {
+        self.viewModel = viewModel
+    }
+
+    public var body: some View {
+        InstUI.BaseScreen(
+            state: viewModel.state,
+            refreshAction: viewModel.refresh
+        ) { _ in
+            LazyVStack(spacing: 0, pinnedViews: .sectionHeaders) {
+                filterCountInfo
+                userFilter
+                courseFilters
+                groupFilters
+            }
+        }
+        .navigationTitle(viewModel.pageTitle)
+        .toolbar {
+            doneButton
+            selectAllButton
+        }
+        .snackBar(viewModel: viewModel.snackbarViewModel)
+        // Without this the refreshable scroll view won't trigger the refresh
+        // because the pull gesture is swallowed by the modal dialog dimiss gesture.
+        // Even if the dismiss itself is disabled the bounce effect will swallow the event.
+        // As a side effect we will see double loading indicators.
+        .refreshable {}
+    }
+
+    private var selectAllButton: some ToolbarContent {
+        ToolbarItem(placement: .topBarTrailing) {
+            Button {
+                viewModel.didTapRightNavButton.send()
+            } label: {
+                if let title = viewModel.rightNavButtonTitle {
```

Comment:
simply `selectAllButtonTitle`, since it is a specific button and we name it like that in Screen property as well?
same for `didTap...`

Reply:
Yeah, I wanted to go in a more generic way but now it seems unnecessary so I renamed these.